src/test/java/org/openmrs/contrib/qaframework/automation/ManageUserAccountSteps.java

```diff
@@ -0,0 +1,136 @@
+package org.openmrs.contrib.qaframework.automation;
+
+import java.util.List;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import org.openmrs.contrib.qaframework.RunTest;
+import org.openmrs.reference.page.ManageUserAccountPage;
+import org.openmrs.reference.page.SystemAdministrationPage;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class ManageUserAccountSteps extends Steps {
+
+	private static final String SYSTEM_ALERT = "Atleast 8 character(s) are required";
+	private SystemAdministrationPage systemAdministrationPage;
+	private ManageUserAccountPage userAccountPage;
+
+	@Before(RunTest.HOOK.SELENIUM_USER_ACCOUNT)
+	public void visitHomePage() {
+		initiateWithLogin();
+	}
+
+	@After(RunTest.HOOK.SELENIUM_USER_ACCOUNT)
+	public void destroy() {
+		quit();
+	}
+
+	@Given("a user clicks on System Administartion app from home page")
+	public void loadSystemAdministrationPage() {
+		systemAdministrationPage = homePage.goToSystemAdministrationPage();
+	}
+
+	@Then("the system loads system administration page")
+	public void systemloadsSystemAdministrationPage() {
+		assertTrue(textExists("System Administration"));
+	}
+
+	@When("a user clicks on Manage accounts app")
+	public void loadManageAccountsPage() {
+		userAccountPage = systemAdministrationPage.goToManageAccounts();
+	}
+
+	@Then("the system loads manage acccount page")
+	public void systemloadsManageAccountsPage() {
+		assertTrue(textExists("Manage Accounts"));
```

Comment:
Thanks @kdaud , Instead of assertTrue() of the existing text, **how about AssertPage(ManageAccountsPage);** since the system is returning a page, i think it would be practically nice asserting a page. Let me know your suggestion.

Reply:
I opted to use rather assert text on the page than asserting a page. Why ?

Below is my experience!

I took almost 3 weeks while working on [RATEST-113](https://issues.openmrs.org/browse/RATEST-113) figuring out why the test was failing. I later discovered that the test was failing because of some pages were taking longer to load and `assertPage() ` would throw exceptions when it could not find the page. I remember one `dev` suggested we rather get ride of the `assertPage()` function but that was not a better alternative rather than figuring the best solution.

I later discovered the solution was to add `waitForPage()` on those pages and then the test passed. However, I went on to read more about loading pages in reference to selenium and discovered that there are some pages that take long to load of which `waitForPage() `does not help because `waitForPage ()` is extending a default selenium time value of 500miles seconds its implementation [here.](https://github.com/openmrs/openmrs-contrib-uitestframework/blob/8838c8eeee749c044f9846bcb0a31a7065db1260/src/main/java/org/openmrs/uitestframework/page/Page.java#L119)

Putting that in mind I rather suggested in the commit to assert the text since it can quickly(easily) be retrieved on the page than asserting the page. Yes some one can say the text is located on the page so its like the same thing, the answer to such an opinion is _NO_ based on the experience. The page can only pass the assertPage() logic when all the elements on the page have been loaded and some page elements take loader to appear on the page due to some technical reasons.

Reply:
> I took almost 3 weeks while working on [RATEST-113](https://issues.openmrs.org/browse/RATEST-113) figuring out why the test was failing. I later discovered that the test was failing because of some pages were taking longer to load and `assertPage() ` would throw exceptions when it could not find the page. I remember one `dev` suggested we rather get ride of the `assertPage()` function but that was not a better alternative rather than figuring out the best solution.

Are you suggesting we can all through textExists() ratherthan assertPage() method

Reply:
> Are you suggesting we can all through use textExists() ratherthan assertPage() method

That is not my suggestion neither! 

You may need to revisit my explanation to get the point clearly. As of now we need to develop codes that are sustainable since we are soon shifting to 3.x RefApp so having auto tests for 2.x RefApp with `code-base` that  can not easily break will reduce the maintenance burden and will help the community in maintaining 2.x RefApp with less efforts.